ballista/scheduler/src/state/task_manager.rs

```diff
@@ -129,26 +126,59 @@ impl ActiveJobQueue {
     }
 }
 
-struct ActiveJobRef<'a> {
-    queue: &'a SegQueue<String>,
-    job: JobInfoCache,
+#[derive(Debug, Clone, Eq, PartialEq, Default)]
+struct ActiveJob {
     job_id: String,
+    running_stage: usize,
+    pending_tasks: usize,
 }
 
-impl<'a> Deref for ActiveJobRef<'a> {
-    type Target = JobInfoCache;
+impl ActiveJob {
+    fn new(job_id: String, graph: &ExecutionGraph) -> Self {
+        let running_stage = graph
+            .running_stages()
+            .iter()
+            .max()
+            .copied()
+            .unwrap_or_default();
+        let pending_tasks = graph.available_tasks();
+        Self {
+            job_id,
+            running_stage,
+            pending_tasks,
+        }
+    }
+}
 
-    fn deref(&self) -> &Self::Target {
-        &self.job
+impl PartialOrd<Self> for ActiveJob {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
     }
 }
 
-impl<'a> Drop for ActiveJobRef<'a> {
-    fn drop(&mut self) {
-        self.queue.push(std::mem::take(&mut self.job_id));
+impl Ord for ActiveJob {
```

Comment:
I think maybe using total remaining tasks might be better in the long run. This would effectively be the same now, but as we add joins and more multi-stage plans the relationship between stage number and tasks remaining becomes murkier. In `ExecutionGraph` we can add a method like

```
    pub fn remaining_tasks(&self) -> usize {
        self.stages
            .values()
            .map(|stage| {
                match stage {
                    ExecutionStage::Running(stage) => stage.available_tasks(),
                    ExecutionStage::Resolved(stage) => stage.partitions,
                    ExecutionStage::UnResolved(stage) => stage.plan.output_partitioning().partition_count(),
                    _ => 0
                }
            })
            .sum()
    }
```

and then just track `remaining_tasks` in `ActiveJob`